internal/cli/api.go

```diff
@@ -0,0 +1,257 @@
+package cli
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/auth0/auth0-cli/internal/ansi"
+	"github.com/auth0/auth0-cli/internal/buildinfo"
+	"github.com/auth0/auth0-cli/internal/display"
+	"github.com/auth0/auth0-cli/internal/iostream"
+	"github.com/auth0/auth0-cli/internal/prompt"
+)
+
+const apiDocsURL = "https://auth0.com/docs/api/management/v2"
+
+var apiFlags = apiCmdFlags{
+	Data: Flag{
+		Name:         "RawData",
+		LongForm:     "data",
+		ShortForm:    "d",
+		Help:         "JSON data payload to send with the request.",
+		IsRequired:   false,
+		AlwaysPrompt: false,
+	},
+	QueryParams: Flag{
+		Name:         "QueryParams",
+		LongForm:     "query",
+		ShortForm:    "q",
+		Help:         "Query params to send with the request.",
+		IsRequired:   false,
+		AlwaysPrompt: false,
+	},
+}
+
+var apiValidMethods = []string{
+	http.MethodGet,
+	http.MethodPost,
+	http.MethodPut,
+	http.MethodPatch,
+	http.MethodDelete,
+}
+
+type (
+	apiCmdFlags struct {
+		Data        Flag
+		QueryParams Flag
+	}
+
+	apiCmdInputs struct {
+		renderer *display.Renderer
+
+		RawMethod      string
+		RawURI         string
+		RawData        string
+		RawQueryParams map[string]string
+		Method         string
+		URL            *url.URL
+		Data           io.Reader
+	}
+)
+
+func apiCmd(cli *cli) *cobra.Command {
+	inputs := apiCmdInputs{
+		renderer: cli.renderer,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "api <method> <uri>",
+		Args:  cobra.RangeArgs(1, 2),
+		Short: "Makes an authenticated HTTP request to the Auth0 Management API",
+		Long: fmt.Sprintf(
+			`Makes an authenticated HTTP request to the Auth0 Management API and prints the response as JSON.
+
+The method argument is optional, and when you don’t specify it, the command defaults to GET for requests without data and POST for requests with data.
```

Comment:
Maybe mention in the documentation that the data can be piped in as well?

Reply:
[`f401dfd` (#531)](https://github.com/auth0/auth0-cli/pull/531/commits/f401dfd09705e3d6960f19f7f9a6333cbcdadba4)